easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -94,6 +95,10 @@ def parse(self, txt):
         """
         txt = self._inject_constants_dict(txt)
         self.parsed_yeb = yaml.load(txt)
+        # make sure dependencies are in a format that can be handled by the EasyConfig class
+        # detecting that auto-converting the list-of-dicts value that results from parsing a .yeb is needed is tricky
+        if 'dependencies' in self.parsed_yeb:
+            self.parsed_yeb['dependencies'] = [to_dependency(d) for d in self.parsed_yeb['dependencies']]
```

Comment:
@Caylo: can you look into a follow-up PR that implements type checking for `dependencies`?

With the new functionality added in here (`is_value_of_type`), it should be fairly easy to get that working.

Then we can remove these two lines, and apply the same conversion trivially to also `builddependencies` and `hiddendependencies` (see http://easybuild.readthedocs.org/en/latest/Writing_easyconfig_files.html#dependencies)